platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/Collectors.java

```diff
@@ -227,13 +240,13 @@ public boolean equals(Object o) {
 
             // We're fine with having weak contract of Iterable/Collection.equals.
             @SuppressWarnings("UndefinedEquals")
-            boolean result = Objects.equal(value, that.value);
-            return result;
+            boolean valuesEqual = Objects.equal(value, that.value);
+            return valuesEqual && Objects.equal(sideEffect, that.sideEffect);
         }
 
         @Override
         public int hashCode() {
-            return Objects.hashCode(value);
+            return Objects.hashCode(value, sideEffect);
```

Comment:
💭 I am not sure if this is ok: as far as I took a look `SideEffect` doesn't have `equals/hashcode` implement.  
I don't know why equals/hashcode is implemented here and if this is in fact an issue

Reply:
>why equals/hashcode is implemented here

These are a part of the implementation of domain object collections. I don't think we're particularly thorough with side effect equality (e.g. side effect providers only have identities too). Guess we're fine for now because side effects aren't used much.

Reply:
Removed that change. I believe that with a specialized `FixedValueCollector` that is used only for plus we can avoid these equals/hashcode. I suspect that since PlusCollector also doesn't have them implemented